src/screens/Customers/AddCustomer.tsx

```diff
@@ -1,38 +1,116 @@
+    hasmeter: false,
+    outstandingBalance: '',
+    meterReadingIds: [],
+    meterReadings: [],
+    paymentIds: [],
+    payments: [],
+    customerUpdateIds: [],
+    customerUpdates: [],
+    totalAmountBilledfromInvoices: 0,
+    totalAmountPaidfromPayments: 0,
+}
 
 const styles = (theme: Theme) =>
   createStyles({
     content: {
       color: theme.palette.text.primary,
     },
+    formControl: {
+      width: '100%',
+    },
   });
 
 interface AddCustomerProps extends RouteComponentProps {
-  classes: { content: string };
+  classes: { content: string, formControl: string};
   location: any;
+  currentSite: SiteRecord;
 }
 
+
 function AddCustomer(props: AddCustomerProps) {
-  const { classes } = props;
+  const { classes, currentSite } = props;
+
+  const [selectedTariffPlan, setSelectedTariffPlan] = useState("");
+  const [customerName, setCustomerName] = useState("");
+  const [meter, setMeter] = useState("");
+  const [meterChecked, setMeterChecked] = useState(false);
+  const [customerInactive, setCustomerInactive] = useState(false);
+
+  const handleSelectTariffPlan = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setSelectedTariffPlan(event.target.value as string);
+  }
+
+  const handleSubmit = (event: React.MouseEvent) => {
+    // Prevent page refresh on submit
+    event.preventDefault();
+
+    const customer = EMPTY_CUSTOMER;
+    customer.name = customerName;
+    customer.meterNumber = parseInt(meter);
+    customer.isactive = !customerInactive;
+    customer.tariffPlansId = selectedTariffPlan;
+    addCustomer(customer);
+    
+    const cust = {
+      ...customer,
+      sitesId: currentSite.rid
+    }
+    
+    createCustomer(cust);
+  }
+
+  const handleNameInput = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setCustomerName(event.target.value as string);
+  }
+
+  const handleMeterInput = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setMeter(event.target.value as string);
+  }
+
+  const tariffPlans = currentSite.tariffPlans;
 
   return (
     <BaseScreen title="Add New Customer" leftIcon="backNav">
-      <form noValidate className={classes.content}>
-        <TextField label={'Name'} />
-        <Checkbox label={'Select if customer is inactive'} />
-        <Checkbox label={'Meter:'} textField={'meter'} />
-        <TextField label={'Tariff Plan'} />
-        <Button label={'ADD'} />
+      <form noValidate className={classes.content} onSubmit={() => false}>
+        <TextField label={'Name'} id={'name'} primary={true} onChange={handleNameInput} />
+        <FormControl variant="outlined" className={classes.formControl}>
+          <InputLabel id="select-tariff-plan-label">Select Tariff Plan</InputLabel>
+          <Select label={"Select Tariff Plan"} id={'select-tariff-plan'} labelId = "select-tariff-plan-label" onChange={handleSelectTariffPlan}>
+            {tariffPlans.map((plan, index) =>
+                <MenuItem key={index} value={plan.id}>{plan.name}</MenuItem>
```

Comment:
For best practice, index is a bad key. Something unique for each item that's being mapped is best. We can use `plan.id` as key i think.

Source: [here](https://medium.com/wesionary-team/using-index-as-a-key-is-an-anti-pattern-in-react-8e5db3aea3a9)